### ISSUE TITLE
open-iscsi delays startup by 2mins

### ISSUE THREAD

**Issue by scyto** [open-iscsi/open-iscsi#299]
I am struggling to understand if this is normal?

```
$ sudo systemd-analyze blame
2min 2.195s open-iscsi.service
     1.991s docker.service
```

I see same 2 min delay when i run `sudo systemctl restart iscsi.service`

- My OS is Debian 11 Bullseye with full apt-update and apt-upgrade
- I am connecting to a synology
- I have tried deleted the cache as per this and it made no difference https://askubuntu.com/questions/988095/how-to-prevent-an-unconnected-iscsi-disk-to-slow-down-my-pc-boot
- I can't read this read this redhat article so don't know if it has anything to offer https://access.redhat.com/solutions/3649231
- I note there is one 120s value in /etc/iscsi/iscsid.conf  `node.session.timeo.replacement_timeout = 120` i haven't tried changing this as i don't know what i am doing
- i amended the open-iscsi systemd file to wait on NetworkManager-wait-online.service - assuming i did it right that didn't help
- i don't remember this taking 2 minutes the first time i set it up and rebooted, so not sure if it was one of the upgrades i did (including kernel) today?

Not sure what to do next?

**Comment by scyto** [comment]
i don't know if this is helpful

```
alex@Docker01:/$ sudo apt show open-iscsi
Package: open-iscsi
Version: 2.1.3-5
Priority: optional
Section: net
Maintainer: Debian iSCSI Maintainers <open-iscsi@packages.debian.org>
Installed-Size: 1,447 kB
Pre-Depends: debconf | debconf-2.0
Depends: debconf (>= 0.5) | debconf-2.0, libc6 (>= 2.17), libisns0, libkmod2 (>= 5~), libmount1 (>= 2.24.2), libopeniscsiusr, libssl1.1 (>= 1.1.1), libsystemd0, udev
Homepage: http://www.open-iscsi.com/
Tag: admin::kernel, devel::lang:c, devel::library, hardware::storage,
 implemented-in::c, interface::commandline, interface::daemon,
 network::client, network::configuration, network::server, protocol::ip,
 role::devel-lib, role::program, scope::application, scope::utility,
 use::storing
Download-Size: 306 kB
APT-Manual-Installed: yes
APT-Sources: http://deb.debian.org/debian bullseye/main amd64 Packages

```

**Comment by gonzoleeman** [comment]
> On Jan 29, 2022, at 12:01 AM, scyto ***@***.***> wrote:
> 
> 
> I am struggling to understand if this is normal?
> 
> $ sudo systemd-analyze blame
> 2min 2.195s open-iscsi.service
>      1.991s docker.service
> I see same 2 min delay when i run sudo systemctl restart iscsi.service
> 
> 
Are you doing this as a test, or do you actually need to restart the service? Just wondering.

> My OS is Debian
Have you filed a bug report there?

> I have tried deleted the cache as per this and it made no difference https://askubuntu.com/questions/988095/how-to-prevent-an-unconnected-iscsi-disk-to-slow-down-my-pc-boot <https://askubuntu.com/questions/988095/how-to-prevent-an-unconnected-iscsi-disk-to-slow-down-my-pc-boot>This is actually a database, not a cache. But ok. You can also remove the node DB entries with: “iscsiadm -m node —op delete”. But I’m not sure how this helps you.
> I can't read this read this redhat article so don't know if it has anything to offer https://access.redhat.com/solutions/3649231 <https://access.redhat.com/solutions/3649231>LOL. If we can’t read the article, why reference it? :)
> I note there is one 120s value in /etc/iscsi/iscsid.conf node.session.timeo.replacement_timeout = 120 i haven't tried changing this as i don't know what i am doing
> i amended the open-iscsi file to wait on NetworkManager-wait-online.service - assuming i did it right that didn't help
That’s probably not wise. The file should be correct already. If not, you should file a bug report with Debian.
> Not sure what to do next?
> 
> 

I’m guessing that this distro has some issues in it’s systemd files. These files are actually different (a bit) between each distribution.

Please show:
- Your iscsid.service and iscsi.service files

If you have not a single entry in your node database, and it still takes 2 minues to start up the iscsi service, something is wrong. :)

— 
Lee Duncan

**Comment by scyto** [comment]
Thanks

1. the restart test was just a test to prove if I could make the same 2min delay issue happen outside of boot, no i don't need to stop and start in production.

2. ahh so i need to file against the distro, got it, i can do that.  I am an ex-windows server guy learning Linux so need to figure out how to find out if they have a package that includes your latest release or not. (ot if it is wise for me to get your tarball and compile?)

this is the command i used to create the node

`sudo iscsiadm -m node -T iqn.2000-01.com.synology:SYN01.Target-1.46dcfe5d51 -p 192.168.1.30:3260 --login`

iscsi.service
```
alex@Docker01:/etc/systemd/system$ cat iscsi.service 
[Unit]
Description=Login to default iSCSI targets
Documentation=man:iscsiadm(8) man:iscsid(8)
Wants=network-online.target remote-fs-pre.target iscsid.service 
After=network-online.target iscsid.service 
Before=remote-fs-pre.target
DefaultDependencies=no
Conflicts=shutdown.target
Before=shutdown.target

[Service]
Type=oneshot
RemainAfterExit=true
# iscsiadm --login will return 21 if no nodes are configured,
# and 15 if a session is alread logged in (which we do not
# consider an error)
SuccessExitStatus=15 21
# iscsiadm will only work if iscsid is running. But we can't use a
# Requires= dependency, since restarts of iscsid would then be
# propagated to this service - and that would cause all sorts of
# mayhem. Therefore, check it in ExecStartPre.
ExecStartPre=/bin/systemctl --quiet is-active iscsid.service
ExecStart=/sbin/iscsiadm -m node --loginall=automatic
ExecStart=/lib/open-iscsi/activate-storage.sh
ExecStop=/lib/open-iscsi/umountiscsi.sh
ExecStop=/bin/sync
ExecStop=/lib/open-iscsi/logout-all.sh

[Install]
WantedBy=sysinit.target
Alias=iscsi.service
```

iscsid.conf, the only thing i changed was the `node.startup` which i changed to automatic
```
alex@Docker01:/etc/iscsi$ cat iscsid.conf 
#
# Open-iSCSI default configuration.
# Could be located at /etc/iscsi/iscsid.conf or ~/.iscsid.conf
#
# Note: To set any of these values for a specific node/session run
# the iscsiadm --mode node --op command for the value. See the README
# and man page for iscsiadm for details on the --op command.
#

######################
# iscsid daemon config
######################
#
# If you want iscsid to start the first time an iscsi tool
# needs to access it, instead of starting it when the init
# scripts run, set the iscsid startup command here. This
# should normally only need to be done by distro package
# maintainers. If you leave the iscsid daemon running all
# the time then leave this attribute commented out.
#
# Default for Fedora and RHEL. (uncomment to activate).
# iscsid.startup = /bin/systemctl start iscsid.socket iscsiuio.socket
#
# Default if you are not using systemd (uncomment to activate)
# iscsid.startup = /usr/bin/service start iscsid

# Check for active mounts on devices reachable through a session
# and refuse to logout if there are any.  Defaults to "No".
# iscsid.safe_logout = Yes

#############################
# NIC/HBA and driver settings
#############################
# open-iscsi can create a session and bind it to a NIC/HBA.
# To set this up see the example iface config file.

#*****************
# Startup settings
#*****************

# To request that the iscsi initd scripts startup a session set to "automatic".
# node.startup = automatic
#
# To manually startup the session set to "manual". The default is manual.
node.startup = automatic

# For "automatic" startup nodes, setting this to "Yes" will try logins on each
# available iface until one succeeds, and then stop.  The default "No" will try
# logins on all available ifaces simultaneously.
node.leading_login = No

# *************
# CHAP Settings
# *************

# To enable CHAP authentication set node.session.auth.authmethod
# to CHAP. The default is None.
#node.session.auth.authmethod = CHAP

# To configure which CHAP algorithms to enable set
# node.session.auth.chap_algs to a comma seperated list.
# The algorithms should be listen with most prefered first.
# Valid values are MD5, SHA1, SHA256, and SHA3-256.
# The default is MD5.
#node.session.auth.chap_algs = SHA3-256,SHA256,SHA1,MD5

# To set a CHAP username and password for initiator
# authentication by the target(s), uncomment the following lines:
#node.session.auth.username = username
#node.session.auth.password = password

# To set a CHAP username and password for target(s)
# authentication by the initiator, uncomment the following lines:
#node.session.auth.username_in = username_in
#node.session.auth.password_in = password_in

# To enable CHAP authentication for a discovery session to the target
# set discovery.sendtargets.auth.authmethod to CHAP. The default is None.
#discovery.sendtargets.auth.authmethod = CHAP

# To set a discovery session CHAP username and password for the initiator
# authentication by the target(s), uncomment the following lines:
#discovery.sendtargets.auth.username = username
#discovery.sendtargets.auth.password = password

# To set a discovery session CHAP username and password for target(s)
# authentication by the initiator, uncomment the following lines:
#discovery.sendtargets.auth.username_in = username_in
#discovery.sendtargets.auth.password_in = password_in

# ********
# Timeouts
# ********
#
# See the iSCSI README's Advanced Configuration section for tips
# on setting timeouts when using multipath or doing root over iSCSI.
#
# To specify the length of time to wait for session re-establishment
# before failing SCSI commands back to the application when running
# the Linux SCSI Layer error handler, edit the line.
# The value is in seconds and the default is 120 seconds.
# Special values:
# - If the value is 0, IO will be failed immediately.
# - If the value is less than 0, IO will remain queued until the session
# is logged back in, or until the user runs the logout command.
node.session.timeo.replacement_timeout = 120

# To specify the time to wait for login to complete, edit the line.
# The value is in seconds and the default is 15 seconds.
node.conn[0].timeo.login_timeout = 15

# To specify the time to wait for logout to complete, edit the line.
# The value is in seconds and the default is 15 seconds.
node.conn[0].timeo.logout_timeout = 15

# Time interval to wait for on connection before sending a ping.
node.conn[0].timeo.noop_out_interval = 5

# To specify the time to wait for a Nop-out response before failing
# the connection, edit this line. Failing the connection will
# cause IO to be failed back to the SCSI layer. If using dm-multipath
# this will cause the IO to be failed to the multipath layer.
node.conn[0].timeo.noop_out_timeout = 5

# To specify the time to wait for abort response before
# failing the operation and trying a logical unit reset edit the line.
# The value is in seconds and the default is 15 seconds.
node.session.err_timeo.abort_timeout = 15

# To specify the time to wait for a logical unit response
# before failing the operation and trying session re-establishment
# edit the line.
# The value is in seconds and the default is 30 seconds.
node.session.err_timeo.lu_reset_timeout = 30

# To specify the time to wait for a target response
# before failing the operation and trying session re-establishment
# edit the line.
# The value is in seconds and the default is 30 seconds.
node.session.err_timeo.tgt_reset_timeout = 30


#******
# Retry
#******

# To specify the number of times iscsid should retry a login
# if the login attempt fails due to the node.conn[0].timeo.login_timeout
# expiring modify the following line. Note that if the login fails
# quickly (before node.conn[0].timeo.login_timeout fires) because the network
# layer or the target returns an error, iscsid may retry the login more than
# node.session.initial_login_retry_max times.
#
# This retry count along with node.conn[0].timeo.login_timeout
# determines the maximum amount of time iscsid will try to
# establish the initial login. node.session.initial_login_retry_max is
# multiplied by the node.conn[0].timeo.login_timeout to determine the
# maximum amount.
#
# The default node.session.initial_login_retry_max is 8 and
# node.conn[0].timeo.login_timeout is 15 so we have:
#
# node.conn[0].timeo.login_timeout * node.session.initial_login_retry_max =
#                                                               120 seconds
#
# Valid values are any integer value. This only
# affects the initial login. Setting it to a high value can slow
# down the iscsi service startup. Setting it to a low value can
# cause a session to not get logged into, if there are distuptions
# during startup or if the network is not ready at that time.
node.session.initial_login_retry_max = 8

################################
# session and device queue depth
################################

# To control how many commands the session will queue set
# node.session.cmds_max to an integer between 2 and 2048 that is also
# a power of 2. The default is 128.
node.session.cmds_max = 128

# To control the device's queue depth set node.session.queue_depth
# to a value between 1 and 1024. The default is 32.
node.session.queue_depth = 32

##################################
# MISC SYSTEM PERFORMANCE SETTINGS
##################################

# For software iscsi (iscsi_tcp) and iser (ib_iser) each session
# has a thread used to transmit or queue data to the hardware. For
# cxgb3i you will get a thread per host.
#
# Setting the thread's priority to a lower value can lead to higher throughput
# and lower latencies. The lowest value is -20. Setting the priority to
# a higher value, can lead to reduced IO performance, but if you are seeing
# the iscsi or scsi threads dominate the use of the CPU then you may want
# to set this value higher.
#
# Note: For cxgb3i you must set all sessions to the same value, or the
# behavior is not defined.
#
# The default value is -20. The setting must be between -20 and 20.
node.session.xmit_thread_priority = -20


#***************
# iSCSI settings
#***************

# To enable R2T flow control (i.e., the initiator must wait for an R2T
# command before sending any data), uncomment the following line:
#
#node.session.iscsi.InitialR2T = Yes
#
# To disable R2T flow control (i.e., the initiator has an implied
# initial R2T of "FirstBurstLength" at offset 0), uncomment the following line:
#
# The defaults is No.
node.session.iscsi.InitialR2T = No

#
# To disable immediate data (i.e., the initiator does not send
# unsolicited data with the iSCSI command PDU), uncomment the following line:
#
#node.session.iscsi.ImmediateData = No
#
# To enable immediate data (i.e., the initiator sends unsolicited data
# with the iSCSI command packet), uncomment the following line:
#
# The default is Yes
node.session.iscsi.ImmediateData = Yes

# To specify the maximum number of unsolicited data bytes the initiator
# can send in an iSCSI PDU to a target, edit the following line.
#
# The value is the number of bytes in the range of 512 to (2^24-1) and
# the default is 262144
node.session.iscsi.FirstBurstLength = 262144

# To specify the maximum SCSI payload that the initiator will negotiate
# with the target for, edit the following line.
#
# The value is the number of bytes in the range of 512 to (2^24-1) and
# the defauls it 16776192
node.session.iscsi.MaxBurstLength = 16776192

# To specify the maximum number of data bytes the initiator can receive
# in an iSCSI PDU from a target, edit the following line.
#
# The value is the number of bytes in the range of 512 to (2^24-1) and
# the default is 262144
node.conn[0].iscsi.MaxRecvDataSegmentLength = 262144

# To specify the maximum number of data bytes the initiator will send
# in an iSCSI PDU to the target, edit the following line.
#
# The value is the number of bytes in the range of 512 to (2^24-1).
# Zero is a special case. If set to zero, the initiator will use
# the target's MaxRecvDataSegmentLength for the MaxXmitDataSegmentLength.
# The default is 0.
node.conn[0].iscsi.MaxXmitDataSegmentLength = 0

# To specify the maximum number of data bytes the initiator can receive
# in an iSCSI PDU from a target during a discovery session, edit the
# following line.
#
# The value is the number of bytes in the range of 512 to (2^24-1) and
# the default is 32768
# 
discovery.sendtargets.iscsi.MaxRecvDataSegmentLength = 32768

# To allow the targets to control the setting of the digest checking,
# with the initiator requesting a preference of enabling the checking, uncomment# one or both of the following lines:
#node.conn[0].iscsi.HeaderDigest = CRC32C,None
#node.conn[0].iscsi.DataDigest = CRC32C,None
#
# To allow the targets to control the setting of the digest checking,
# with the initiator requesting a preference of disabling the checking,
# uncomment one or both of the following lines:
#node.conn[0].iscsi.HeaderDigest = None,CRC32C
#node.conn[0].iscsi.DataDigest = None,CRC32C
#
# To enable CRC32C digest checking for the header and/or data part of
# iSCSI PDUs, uncomment one or both of the following lines:
#node.conn[0].iscsi.HeaderDigest = CRC32C
#node.conn[0].iscsi.DataDigest = CRC32C
#
# To disable digest checking for the header and/or data part of
# iSCSI PDUs, uncomment one or both of the following lines:
#node.conn[0].iscsi.HeaderDigest = None
#node.conn[0].iscsi.DataDigest = None
#
# The default is to never use DataDigests or HeaderDigests.
#

# For multipath configurations, you may want more than one session to be
# created on each iface record.  If node.session.nr_sessions is greater
# than 1, performing a 'login' for that node will ensure that the
# appropriate number of sessions is created.
node.session.nr_sessions = 1

# When iscsid starts up it recovers existing sessions, if possible.
# If the target for a session has gone away when this occurs, the
# iscsid daemon normally tries to reestablish each session,
# in succession, in the background, by trying again every two
# seconds, until all sessions are restored. This configuration
# variable can limits the number of retries for each session.
# For example, setting reopen_max=150 would mean that each session
# recovery was limited to about five minutes.
# 
node.session.reopen_max = 0

#************
# Workarounds
#************

# Some targets like IET prefer after an initiator has sent a task
# management function like an ABORT TASK or LOGICAL UNIT RESET, that
# it does not respond to PDUs like R2Ts. To enable this behavior uncomment
# the following line (The default behavior is Yes):
node.session.iscsi.FastAbort = Yes

# Some targets like Equalogic prefer that after an initiator has sent
# a task management function like an ABORT TASK or LOGICAL UNIT RESET, that
# it continue to respond to R2Ts. To enable this uncomment this line
# node.session.iscsi.FastAbort = No

# To prevent doing automatic scans that would add unwanted luns to the system
# we can disable them and have sessions only do manually requested scans.
# Automatic scans are performed on startup, on login, and on AEN/AER reception
# on devices supporting it.  For HW drivers all sessions will use the value
# defined in the configuration file.  This configuration option is independent
# of scsi_mod scan parameter. (The default behavior is auto):
node.session.scan = auto
```

**Comment by gonzoleeman** [comment]
> On Jan 29, 2022, at 11:40 AM, scyto ***@***.***> wrote:
> 
> 
> Thanks
> 
> the restart test was just a test to prove if I could make the same 2min delay issue happen outside of boot, no i don't need to stop and start in production.
> 
> ahh so i need to file against the distro, got it, i can do that. I am an ex-windows server guy learning Linux so need to figure out how to find out if they have a package that includes your latest release or not. (ot if it is wise for me to get your tarball and compile?)
> 
> 
This may be an issue your distro has a fix for, or has an update that fixes. They may know what the problem is already, if other’s have filed a bug report as well. And there’s no way they are going to fix it if you don’t let them know it’s broken, IMHO. But it doesn’t hurt to ask here, as well.
> 
> this is the command i used to create the node
> 
> sudo iscsiadm -m node -T iqn.2000-01.com.synology:SYN01.Target-1.46dcfe5d51 -p 192.168.1.30:3260 --login
> 
> 
Well, no, actually that command accesses an existing node. The nodes are created during discovery. So somebody had to run something like “iscsiadm -m discovery -t st 192.168.1.30”.

What does “iscsiadm -m node” show? And how about “iscsiadm -m node -P1”. Also, what about “iscsiadm -m node —op show | fgrep startup”?
> iscsi.service
> 
> ***@***.***:/etc/systemd/system$ cat iscsi.service 
> [Unit]
> Description=Login to default iSCSI targets
> Documentation=man:iscsiadm(8) man:iscsid(8)
> Wants=network-online.target remote-fs-pre.target iscsid.service 
> After=network-online.target iscsid.service 
> Before=remote-fs-pre.target
> DefaultDependencies=no
> Conflicts=shutdown.target
> Before=shutdown.target
> 
> [Service]
> Type=oneshot
> RemainAfterExit=true
> # iscsiadm --login will return 21 if no nodes are configured,
> # and 15 if a session is alread logged in (which we do not
> # consider an error)
> SuccessExitStatus=15 21
> # iscsiadm will only work if iscsid is running. But we can't use a
> # Requires= dependency, since restarts of iscsid would then be
> # propagated to this service - and that would cause all sorts of
> # mayhem. Therefore, check it in ExecStartPre.
> ExecStartPre=/bin/systemctl --quiet is-active iscsid.service
> ExecStart=/sbin/iscsiadm -m node --loginall=automatic
> ExecStart=/lib/open-iscsi/activate-storage.sh
> ExecStop=/lib/open-iscsi/umountiscsi.sh
> ExecStop=/bin/sync
> ExecStop=/lib/open-iscsi/logout-all.sh

Ya, as I suspected, this is totally different than (1) the upstream values in these files, and (2) the distro I work on (SUSE).

In reality, the only purpose of this service is to log into targets you want logins to, at system startup time, and to logout of said targets at system shutdown time.

I have no idea what “activate-storage.sh” is, since Debian evidently hasn’t shared that extension. :) But, as far as I know, it shouldn’t be needed if things are ordered correctly via systemd. (Easier said than done.)
> 
> [Install]
> WantedBy=sysinit.target
> Alias=iscsi.service
> iscsid.conf, the only thing i changed was the node.startup which i changed to automatic
> 
> 
Ah, ok. So you evidently do want your iscsi nodes to be automatically logged into when they system comes up — and logged out of when the system shuts down. Ok. So blowing away your node database is probably a “bad thing”.
> ***@***.***:/etc/iscsi$ cat iscsid.conf 
> #
> # Open-iSCSI default configuration.
> # Could be located at /etc/iscsi/iscsid.conf or ~/.iscsid.conf
> #
> … (removing the whole iscsid.conf file, which wasn’t really needed)
> —
> 

Please show iscsid.service, and iscsid.socket, if the socket file exists.

I assume you are not using iscsiuio.socket/iscsiuio.service (unless you have a hardware CNA accelerator card).

— 
Lee Duncan

**Comment by gonzoleeman** [comment]
Any progress? Have you filed an issue with your distro?

**Comment by scyto** [comment]
apologies no, we can close this here until that's done if you like?

**Comment by scyto** [comment]
`I assume you are not using iscsiuio.socket/iscsiuio.service (unless you have a hardware CNA accelerator card).`
nope this is all inside VMs :-)

**Comment by scyto** [comment]
not sure i get what you mean by show iscsid.service?

**Comment by gonzoleeman** [comment]
> not sure i get what you mean by show iscsid.service?

I mean: "cat /usr/lib/systemd/system/iscsid.service" :)

**Comment by scyto** [comment]
Thanks, i couldn't find it :-)

```
[Unit]
Description=iSCSI initiator daemon (iscsid)
Documentation=man:iscsid(8)
Wants=network-online.target remote-fs-pre.target
Before=remote-fs-pre.target
After=network.target network-online.target
DefaultDependencies=no
Conflicts=shutdown.target
Before=shutdown.target

[Service]
Type=forking
PIDFile=/run/iscsid.pid
ExecStartPre=/lib/open-iscsi/startup-checks.sh
ExecStart=/sbin/iscsid
ExecStop=/sbin/iscsiadm -k 0 2

[Install]
WantedBy=sysinit.target
```

**Comment by gonzoleeman** [comment]
Well, without being able to look at your system, it's hard for me to tell what is adding 2 minutes, but I'm pretty sure it's not open-iscsi itself.

You might try enabling iscsid debugging, as well as adding debugging to the iscsiadm calls, to see what is taking so long.

If it was me, I might try commenting out the "ExecStartPre=" line, just to see if that is the problem. I might also remove "network.target" string from the "After=" line, since network-online.target covers that. But that probably isn't the issue.

I might also try removing your iscsi target, to see if logging into it is adding the delay. Or you could just disable iscsi.service (e.g. "systemctl disable iscsi.service; systemctl stop iscsi.service".

To add debugging, add "-d4" to the iscsid and iscsiadm commands in the service file, then reload the systemd units (or reboot).

**Comment by scyto** [comment]
Thanks.  I think you are right.

I enabled the -d4 - not sure where that adds debug logs to?

In /var/log/messages  and dmesg it appears the iscsi device is logged into and attached within the firs 3.6 seconds of boot.

The large delay happens between keepalived starting and the EXT4-fs mounting.
The only thing in that is hyper-v dynamic memory - my next test is set fixed memory and turn off some of the integrations (poke'n'hope troubleshooting) - not sure how/why the iscsi services are getting the 'blame'?

please lert me know if there are other logs i should be looking at,



```
[    2.165353] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    3.382826] scsi host1: iSCSI Initiator over TCP/IP
[    3.586213] scsi 1:0:0:1: Direct-Access     SYNOLOGY iSCSI Storage    4.0  PQ: 0 ANSI: 5
[    3.590912] sd 1:0:0:1: Attached scsi generic sg3 type 0
[    3.593288] sd 1:0:0:1: [sdc] 209715200 512-byte logical blocks: (107 GB/100 GiB)
[    3.593556] sd 1:0:0:1: [sdc] Write Protect is off
[    3.593557] sd 1:0:0:1: [sdc] Mode Sense: 43 00 10 08
[    3.594556] sd 1:0:0:1: [sdc] Write cache: enabled, read cache: enabled, supports DPO and FUA
[    3.595905] sd 1:0:0:1: [sdc] Optimal transfer size 16384 logical blocks > dev_max (8192 logical blocks)
[    3.616309] sd 1:0:0:1: [sdc] Attached SCSI disk
[   49.517375] hv_balloon: Max. dynamic memory size: 4096 MB
[  125.777201] EXT4-fs (sdc): mounted filesystem with ordered data mode. Opts: (null)
[  125.780656] FS-Cache: Loaded
[  125.819471] FS-Cache: Netfs 'nfs' registered for caching
```


```
Feb 17 13:17:09 Docker01 Keepalived_vrrp[540]: Opening file '/etc/keepalived/keepalived.conf'.
Feb 17 13:17:09 Docker01 Keepalived_vrrp[540]: Registering gratuitous ARP shared channel
Feb 17 13:17:09 Docker01 Keepalived_vrrp[540]: Delaying startup for 5 seconds
Feb 17 13:17:09 Docker01 kernel: [    3.382826] scsi host1: iSCSI Initiator over TCP/IP
Feb 17 13:17:09 Docker01 kernel: [    3.586213] scsi 1:0:0:1: Direct-Access     SYNOLOGY iSCSI Storage    4.0  PQ: 0 ANSI: 5
Feb 17 13:17:09 Docker01 kernel: [    3.590912] sd 1:0:0:1: Attached scsi generic sg3 type 0
Feb 17 13:17:09 Docker01 kernel: [    3.593288] sd 1:0:0:1: [sdc] 209715200 512-byte logical blocks: (107 GB/100 GiB)
Feb 17 13:17:09 Docker01 kernel: [    3.593556] sd 1:0:0:1: [sdc] Write Protect is off
Feb 17 13:17:09 Docker01 kernel: [    3.594556] sd 1:0:0:1: [sdc] Write cache: enabled, read cache: enabled, supports DPO and FUA
Feb 17 13:17:09 Docker01 kernel: [    3.595905] sd 1:0:0:1: [sdc] Optimal transfer size 16384 logical blocks > dev_max (8192 logical blocks)
Feb 17 13:17:09 Docker01 kernel: [    3.616309] sd 1:0:0:1: [sdc] Attached SCSI disk
Feb 17 13:17:14 Docker01 Keepalived_vrrp[540]: (VI_1) Entering BACKUP STATE (init)
Feb 17 13:17:14 Docker01 Keepalived_vrrp[540]: (VI_1) received lower priority (50) advert from 192.168.1.42 - discarding
Feb 17 13:17:15 Docker01 Keepalived_vrrp[540]: (VI_1) received lower priority (50) advert from 192.168.1.42 - discarding
Feb 17 13:17:16 Docker01 Keepalived_vrrp[540]: (VI_1) received lower priority (50) advert from 192.168.1.42 - discarding
Feb 17 13:17:17 Docker01 Keepalived_vrrp[540]: (VI_1) received lower priority (50) advert from 192.168.1.42 - discarding
Feb 17 13:17:18 Docker01 Keepalived_vrrp[540]: (VI_1) Entering MASTER STATE
Feb 17 13:18:02 Docker01 kernel: [   49.517375] hv_balloon: Max. dynamic memory size: 4096 MB
Feb 17 13:19:18 Docker01 kernel: [  125.777201] EXT4-fs (sdc): mounted filesystem with ordered data mode. Opts: (null)
Feb 17 13:19:18 Docker01 kernel: [  125.780656] FS-Cache: Loaded
Feb 17 13:19:18 Docker01 kernel: [  125.819471] FS-Cache: Netfs 'nfs' registered for caching
Feb 17 13:19:18 Docker01 kernel: [  125.822089] nfs: Deprecated parameter 'intr'
```

**Comment by scyto** [comment]
I have never reported a bug via /  used a mailing list but i will try.
I don't see any obvious outstanding debian bugs https://bugs.debian.org/cgi-bin/pkgreport.cgi?dist=unstable;package=open-iscsi  (but then i have no clue what i am talking about)

also captured this if it is helpful (attached)
